datatypes/src/primitives/spatial_partition.rs

```diff
@@ -70,12 +70,21 @@ impl SpatialPartition2D {
     /// Create a partition from a bbox by snapping to the next pixel
     /// The resulting partition is not equivalent to the bbox but contains it
     pub fn with_bbox_and_resolution(bbox: BoundingBox2D, resolution: SpatialResolution) -> Self {
+        const EPSILON: f64 = 0.000_001;
```

Comment:
maybe add a todo to get rid of this epsilon and just snap the the next pixel, once spatial partitions are reworked.